### PR TITLE
VMO-6059 - feature/VMO-6059/message-block-ui-integration

### PR DESCRIPTION
**Pull request by bulatgab**

### CODE REVIEW

**Review by object-required — APPROVED** [review]

**Comment by safydy-r** [comment]
@bulatgab , when checking this branch, I'm noticing there is a pbm on the validation:
- all the validation list is gone when we provide a block title
I think we need fix this in current ticket even we haven't noticed during ticket creation.
vmo-6059 - issue on validation

- [ ] fix validation as described above

File: .eslintrc.js [hunk 1]
```diff
@@ -360,6 +360,7 @@ module.exports = {
     'no-else-return': ['off'],
     'spaced-comment': 'off',
     'line-comment-position': ['error', {position: 'above'}],
+    'arrow-parens': 'off',
```
Comment (bulatgab) [inline .eslintrc.js:363]: This allows us to get rid of parenthesis around a single parameter in arrow functions, for example:
```
resources.some((resource) => ...)
```
becomes
```
resources.some(resource => ...)
```

File: src/components/interaction-designer/toolbar/ErrorNotifications.vue [hunk 2]
```diff
@@ -111,26 +111,19 @@ export class ErrorNotifications extends mixins(Routes, Lang) {
     return pickBy(this.validationStatuses, (value: IValidationStatus, key) => key.includes('block/') && blocksMap[key])
   }
 
-  get resourceValidationStatusesForActiveFlow(): Dictionary<IValidationStatus> {
-    return pickBy(this.validationStatuses, (status, key) => {
-      const resourceUuid = replace(key, 'resource/', '')
-      return this.resourceUuidsOnActiveFlow.includes(resourceUuid) && !get(this.validationStatuses, `resource/${resourceUuid}`)?.isValid
-    })
-  }
-
```
Comment (bulatgab) [inline src/components/interaction-designer/toolbar/ErrorNotifications.vue:120]: I think the issue here was using `this` in an arrow function in a class's method which gave it a wrong scope (`this` was undefined)

File: src/components/interaction-designer/toolbar/ErrorNotifications.vue [hunk 3]
```diff
@@ -111,26 +111,19 @@ export class ErrorNotifications extends mixins(Routes, Lang) {
...
-      if (has(this.resourceValidationStatusesForActiveFlow, `resource/${uuids[i]}`)) {
-        return true
-      }
-    }
-    return false
+    return uuids.some(uuid => {
+      const isValid = this.validationStatuses[`resource/${uuid}`]?.isValid
+      const doesBelongToActiveFlow = this.activeFlow?.resources.some(resource => resource.uuid === uuid)
+
+      return isValid === false && doesBelongToActiveFlow
```
Comment (bulatgab) [inline src/components/interaction-designer/toolbar/ErrorNotifications.vue:125]: `isValid` can be not only boolean, so a strict comparison with `false` is required

**Comment by bulatgab** [comment]
Hi @safydy-r,
Thanks for pointing out that issue!
I've fixed it in this commit and left some comments:
https://github.com/FLOIP/flow-builder/pull/258/commits/92c9f5187f054056e076d00abf1425819c8111a0

File: src/components/interaction-designer/toolbar/ErrorNotifications.vue (hunk 3 above)
Comment (safydy-r) [reply src/components/interaction-designer/toolbar/ErrorNotifications.vue:125]: Thanks for this, I like the usage of `some()`, which make an early return:
> As soon as one `true` is returned, `.some()` will itself return `true`

**Review by safydy-r — APPROVED** [review]
Thanks @bulatgab for fixing this 💯

**Pull request merged**